suite/tests/api/ir_x86.c

```diff
@@ -299,6 +299,88 @@ test_all_opcodes_3_avx512_vex(void *dc)
 #    undef INCLUDE_NAME
 }
 
+static void
+test_all_opcodes_mask_2_avx512_evex(void *dc)
+{
+#    define INCLUDE_NAME "ir_x86_mask_2args_avx512_evex.h"
+#    define OPCODE_FOR_CREATE(name, opc, icnm, flags, arg1, arg2, arg3)             \
```

Comment:
Optional: I know there is existing splitting but it seems unnecessary -- this could be shared just with a new INCLUDE_NAME and new include of ir_x86_all_opc.h in an existing test_all_opcodes_3() routine.

Reply:
We might want to split even more. There are so many instructions (see i#1312), the lists will become huge!

Reply:
I don't follow which approach you are suggesting: if there are many, it seems that sharing code will be good, but you're also saying to "split even more" and to have many separate functions.  I don't see any big advantage to separate functions when the contents are auto-generated like this: I would say let's combine them and avoid duplicating this boilerplate when the number of args is the same (or could use vararg macros).

Reply:
There will be a lot of avx-512 related tests like this, the files would become very big. Also I wanted to make mask + 2 args (now called mask_3args as per above) distinct from 3 args. What do you mean by auto-generated? I wish it was! The _avx version has its own function as well. I am however not feeling strongly about this. Did you want to merge them all into one function, i.e. _avx, _avx512 etc. into test_all_opcodes_3?

Reply:
The suggestion is trying to get rid of the duplicated OPCODE_FOR_CREATE define.  How about defining it once outside of the 3-arg functions instead and keeping separate functions?  I.e., it's defined just once for each # of args and shared among all such functions.

Reply:
Since the PR here is already big, what about I set up a PR right after that re-factors ir_x86.c into OPCODE_FOR_CREATE_1ARG/2ARGS/3ARGS etc.?

Reply:
OK